src/kani-driver/src/call_cbmc.rs

```diff
@@ -127,3 +133,192 @@ impl KaniSession {
         args
     }
 }
+
+/// Solve Unwind Value from conflicting inputs of unwind values. (--unwind, annotation-unwind, --harness-unwind)
+pub fn resolve_unwind_value(args: &KaniArgs, harness_metadata: &HarnessMetadata) -> Option<u32> {
+    // Check for which flag is being passed and prioritize extracting unwind from the
+    // respective flag/annotation.
+    if let Some(harness_unwind) = args.harness_unwind {
+        Some(harness_unwind)
+    } else if let Some(annotation_unwind) = harness_metadata.unwind_value {
+        Some(annotation_unwind)
+    } else if let Some(default_unwind) = args.unwind {
+        Some(default_unwind)
+    } else {
+        None
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::args;
+    use crate::metadata::{find_proof_harness, mock_proof_harness};
+    use structopt::StructOpt;
+
+    use super::*;
+
+    #[test]
+    fn check_default_annotation_unwind_resolve() {
+        // Annotation value for unwind takes precedence over unwind value
+        // except when there is no #[kani::unwind()] provided.
+        let args: Vec<OsString> =
+            ["kani", "--no-default-checks", "--unwind", "2"].iter().map(|&s| s.into()).collect();
+        let unwind_args = args::KaniArgs::from_iter(args);
+
+        let harnesses = vec![
```

Comment:
These should just be separate variables, and there's no need for `find_proof_harness` then.